rfcs/0006-merkle-tree/0006-merkle-tree.md

```diff
@@ -102,24 +103,25 @@ Proof gen_proof(Hash tree[], int indexes[]) {
     }
   }
 
-  return Proof::new(size, nodes);
+  add (size-1) for every index in indexes;
+  sort indexes in ascending order by corresponding hash;
+
+  return Proof::new(indexes, nodes);
 }
 ```
 
 ### Algorithm of validation
 
 ```c++
-bool validate_proof(Proof proof, Hash root, Item items[]) {
-  if(proof.size = 0) {
-    return root == H256::zero;
-  }
-  
+bool validate_proof(Proof proof, Hash root, Item items[]) {  
   Queue queue;
-  desending_sort_by_item_index(items);
+  ascending_sort_by_item_hash(items);
   
-  for item in items {
-    queue.push_back((item.hash(), item.index() + Proof.size - 1));
+  for (index,item) in (proof.indexes, items) {
```

Comment:
How to make sure that client knows the items order is same as proof indexes order?

Reply:
We sort items by items' hash, the indexes are also in the same order.